index.js

```diff
@@ -73,16 +75,42 @@ function getMainArgs() {
 }
 
 const protoKey = '__proto__';
-function storeOptionValue(options, longOption, value, result) {
-  const optionConfig = options[longOption] || {};
+
+function storeOption({
+  strict,
+  options,
+  result,
+  longOption,
+  shortOption,
+  optionValue,
+}) {
+  const hasOptionConfig = ObjectHasOwn(options, longOption);
+
+  const optionConfig = hasOptionConfig ? options[longOption] : {};
+
+  if (strict) {
+    if (!hasOptionConfig) {
+      throw new ERR_UNKNOWN_OPTION(shortOption == null ? `--${longOption}` : `-${shortOption}`);
+    }
+
+    const shortOptionErr = optionConfig.short ? `-${optionConfig.short}, ` : '';
```

Comment:
It was validated as a single character on entry, so no need for extra checks here.

Reply:
The initial validation only checks own properties of the option config. Couldn't immediatly think of a way to confirm, but came up with this demo:
![image](https://user-images.githubusercontent.com/32409546/163031709-d067a215-4c54-4ddb-9076-2c9f17992485.png)

Not sure if this opens up any bugs or areas for malicious behavior, but might as well update the condition to:
```js
const shortOptionErr = ObjectHasOwn(optionConfig, 'short') ? `-${optionConfig.short}, ` : '';
```

Reply:
https://github.com/pkgjs/parseargs/pull/74/commits/12fd0e6d58fda45d49f2d310cf4ec0090e03fcbe

Reply:
Arg, you are right! (Sorry, the validation wasn't doing what I thought.) 

I assume the high level goal to behave the same as if the prototype pollution was not present?

That makes me question lots more of the code though. The options bag has introduced lots of property access:
- same issue for `multiple`
- and `type`, lots including utils. Although that might be ok if the next issue was fixed.
- the validation of `type` at the top of `parseArgs` does not test it is not from prototype (from my recent change to make type required!)
- the destructing function parameters of both `parseArgs` and (now) `storeOption` receive properties from prototype if not specified by caller.

We could copy the input arguments into a safe(r) object before using them, by specifying all expected properties. Is this a pattern used in other node code? I suspect the usual pattern is just be paranoid everywhere because easier to be confident local code is being paranoid enough?!

I suggest only fix up the code introduced in this PR though. Get `strict` functional and separately have an more-paranoia-more-robust revisit!

Reply:
@bcoe @ljharb 
If my reading of this is correct, we are not currently robust against prototype pollution. Can we improve this after first version of a prototype feature?

(I have some ideas about how to improve it without littering the code and can have a go tonight, but clock is ticking.)

Reply:
I think the bigger concern for prototype pollution is untrusted input to the CLI modifying the prototype, vs., the user having mucked with __proto__ in their application, and this affecting our parse.

This makes me think that the bigger security concern would be if we ever supported dot properties; as @shadowspawn has mentioned.